pandas/core/series.py

```diff
@@ -486,6 +486,8 @@ def _init_dict(
             # fastpath for Series(data=None). Just use broadcasting a scalar
             # instead of reindexing.
             values = na_value_for_dtype(pandas_dtype(dtype), compat=False)
+            if values is None:
```

Comment:
use construct_1d_arraylike_from_scalar(

Reply:
Without the current changes on line 489 we won't ever reach `construct_1d_arraylike_from_scalar` due to the infinite recursion.  If `data` is None. then the Series constructor sets `data = {}` and `_init_dict` is called. When the index is not None `_init_dict` sets `data` to the NA value for the dtype and calls the Series constructor.  When this NA value is None we are back at step 1 and we get the infinite recursion.

Reply:
i think jreback meant to use construct_1d_arraylike_from_scalar on L490

Reply:
like this?
```
dtype_ = pandas_dtype(dtype)
values = na_value_for_dtype(dtype_, compat=False)
if values is None:
    values = construct_1d_arraylike_from_scalar(values, len(index), dtype_)
```

Reply:
yes